pkg/policy/tree.go

```diff
@@ -199,12 +205,21 @@ func (t *Tree) Add(path string, node *Node) (bool, error) {
 			if modified, err = t.Root.Merge(node); err != nil {
 				return false, err
 			}
+			// If modified we need to resolve the tree's root
```

Comment:
Note the `ResolveTree()` at the end is for `node` as this `ResolveTree()` is for `t.Root`, the merge merges the `node` into the `t.Root` and doesn't mean that `node`==`t.Root`.